source/dofs/dof_tools.cc

```diff
@@ -1358,6 +1358,56 @@ namespace DoFTools
 
 
 
+  template <int dim, int spacedim>
+  std::map<typename DoFHandler<dim - 1, spacedim>::active_cell_iterator,
+           std::pair<typename DoFHandler<dim, spacedim>::active_cell_iterator,
+                     unsigned int>>
+  map_boundary_to_bulk_dof_iterators(
+    const std::map<typename Triangulation<dim - 1, spacedim>::cell_iterator,
+                   typename Triangulation<dim, spacedim>::face_iterator>
+      &                                  c1_to_c0,
+    const DoFHandler<dim, spacedim> &    c0_dh,
+    const DoFHandler<dim - 1, spacedim> &c1_dh)
+  {
+    // This is the returned object: a map of codimension-1 active dof cell
+    // iterators to codimension-0 cells and face indices
+    std::map<typename DoFHandler<dim - 1, spacedim>::active_cell_iterator,
+             std::pair<typename DoFHandler<dim, spacedim>::active_cell_iterator,
+                       unsigned int>>
+      c1_to_c0_cells_and_faces;
+
+    // Shortcut if there are no faces to check
+    if (c1_to_c0.empty())
+      return c1_to_c0_cells_and_faces;
+
+    // This is the partial inverse of the map passed as input, for dh
+    std::map<typename Triangulation<dim, spacedim>::face_iterator,
+             typename DoFHandler<dim - 1, spacedim>::active_cell_iterator>
+      c0_to_c1;
+
+    // map volume mesh face -> codimension 1 dof cell
+    for (const auto &c1_cell : c1_dh.active_cell_iterators())
+      if (c1_to_c0.find(c1_cell) != c1_to_c0.end())
+        c0_to_c1[c1_to_c0.at(c1_cell)] = c1_cell;
```

Comment:
I am a bit add odds with this statement because it incurs a huge quadratic complexity. What about simply iterating over the c1_to_c0 map (which is order n) and then inserting all entries into the inverse map? (Untested:)
```
for (const auto &[c1_cell, c0_cell] : c1_to_c0)
  {
    Assert(c0_to_c1.find(c0_cell) == c0_to_c1.end(), ExcInternalError());
    c0_to_c1[c0_cell] = c1_cell;
  }
```

Reply:
They are of different types. I guess we can do that with cell->as_dof_handler_iterator(dh) though. I'll try it out.